broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -40,39 +41,84 @@ class ServiceBrokerApiController extends FabrikBaseController {
 
   getCatalog(req, res) {
     /* jshint unused:false */
-    res.status(200).json(this.fabrik.getPlatformManager(req.params.platform).getCatalog(catalog));
+    res.status(CONST.HTTP_STATUS_CODE.OK).json(this.fabrik.getPlatformManager(req.params.platform).getCatalog(catalog));
   }
 
   putInstance(req, res) {
+    if (config.enable_service_fabrik_v2) {
+      return this.putInstanceV2(req, res);
+    }
+    return this.putInstanceV1(req, res);
+  }
+
+  putInstanceV1(req, res) {
     const params = _.omit(req.body, 'plan_id', 'service_id');
 
     function done(result) {
-      let statusCode = 201;
+      let statusCode = CONST.HTTP_STATUS_CODE.CREATED;
       const body = {
         dashboard_url: req.instance.dashboardUrl
       };
       if (req.instance.async) {
-        statusCode = 202;
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
         body.operation = utils.encodeBase64(result);
       }
       res.status(statusCode).send(body);
     }
 
     function conflict(err) {
       /* jshint unused:false */
-      res.status(409).send({});
+      res.status(CONST.HTTP_STATUS_CODE.CONFLICT).send({});
     }
 
     req.operation_type = CONST.OPERATION_TYPE.CREATE;
-    this.validateRequest(req, res);
 
     return Promise
       .try(() => req.instance.create(params))
       .then(done)
       .catch(ServiceInstanceAlreadyExists, conflict);
   }
 
+  putInstanceV2(req, res) {
+    const params = _.omit(req.body, 'plan_id', 'service_id');
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.CREATED;
+      const body = {
+        dashboard_url: req.instance.dashboardUrl
+      };
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      }
+      res.status(statusCode).send(body);
+    }
+
+    function conflict(err) {
+      /* jshint unused:false */
+      res.status(CONST.HTTP_STATUS_CODE.CONFLICT).send({});
+    }
+
+    req.operation_type = CONST.OPERATION_TYPE.CREATE;
+
+    return req.instance.create(params)
+      .then(done)
+      // Release lock in case of error: catch and throw
+      .catch(err => {
```

Comment:
not locking, but just unlocking does not seem right. Code readability is affected. May be have a wrapper method in BaseController named something like 'handleByLockingResource' and this method is mapped in the route instead of 'handler' method of BaseController or 'handler' method can be enhanced to take in a optional boolean param named `lockResource`. So in this function resource is locked before invoking ` fn.call(this, req, res)` and unlocked prior to retuning back from this method.  
This is just one of the ways, other ways could be how EventLogInterceptor has been implemented. What i meant yday was that normal code should not be littered with locking code.